GitHubCodespaces.ps1

```diff
@@ -0,0 +1,950 @@
+                $description = "Getting codespaces for $OrganizationName"
+            }
+            else
+            {
+                $telemetryProperties['UserName'] = Get-PiiSafeString -PlainText $UserName
+                $uriFragment = "orgs/$OrganizationName/members/$UserName/codespaces"
+                $description = "Getting codespaces for $OrganizationName"
+            }
+
+            break
+        }
+
+        { $_ -in ('Elements', 'Uri') }
+        {
+            $elements = Resolve-RepositoryElements
+            $OwnerName = $elements.ownerName
+            $RepositoryName = $elements.repositoryName
+
+            $telemetryProperties['OwnerName'] = Get-PiiSafeString -PlainText $OwnerName
+            $telemetryProperties['RepositoryName'] = Get-PiiSafeString -PlainText $RepositoryName
+
+            $uriFragment = "repos/$OwnerName/$RepositoryName/codespaces"
+            $description = "Getting $OwnerName/$RepositoryName/codespaces"
+
+            break
+        }
+    }
+
+    $params = @{
+        UriFragment = $uriFragment
+        Description = $description
+        AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+        TelemetryProperties = $telemetryProperties
+    }
+
+    $result = Invoke-GHRestMethodMultipleResult @params
+    if ($null -ne $result.codespaces)
+    {
+        $result = $result.codespaces
+    }
+
+    return ($result | Add-GitHubCodespaceAdditionalProperties)
+}
+
+function New-GitHubCodespace
+{
+    <#
+    .SYNOPSIS
+        Creates a codespace.
+
+    .DESCRIPTION
+        Creates a codespace.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OwnerName
+        Owner of the Codespace.
+        If not supplied here, the DefaultOwnerName configuration property value will be used.
+
+    .PARAMETER RepositoryName
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the Codespace.
+        The OwnerName and CodespaceName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER PullRequest
+        The pull request number for this codespace.
+
+    .PARAMETER RepositoryId
+        The ID for a Repository.  Only applicable when creating a codespace for the current authenticated user.
+
+    .PARAMETER Ref
+        Git ref (typically a branch name) for this codespace
+
+    .PARAMETER ClientIp
+        IP for geo auto-detection when proxying a request.
+
+    .PARAMETER DevContainerPath
+        Path to devcontainer.json config to use for this codespace.
+
+    .PARAMETER DisplayName
+        Display name for this codespace
+
+    .PARAMETER Geo
+        The geographic area for this codespace.
+        Assigned by IP if not provided.
+        Can be one of: EuropeWest, SoutheastAsia, UsEast, UsWest
```

Comment:
This seems risky.  I deliberately avoided it because geo is relatively new.  I'm worried the ValidateSet might be debilitatingly restrictive should they change or add options.  But, I suppose it isn't terribly difficult to change if needed.